From cd014daec9ff329b6fc8c004c8c2d82d04389585 Mon Sep 17 00:00:00 2001
From: yury <yury.korolev@gmail.com>
Date: Tue, 24 Aug 2010 18:27:49 +0400
Subject: [PATCH] Remove unnecessary code from UrlHelper#link_to.

convert_options_to_data_attributes always returns not nil stringified html_options
---
 actionpack/lib/action_view/helpers/url_helper.rb |    9 ++-------
 1 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/actionpack/lib/action_view/helpers/url_helper.rb b/actionpack/lib/action_view/helpers/url_helper.rb
index a5c6718..4a18222 100644
--- a/actionpack/lib/action_view/helpers/url_helper.rb
+++ b/actionpack/lib/action_view/helpers/url_helper.rb
@@ -235,13 +235,8 @@ module ActionView
           html_options = convert_options_to_data_attributes(options, html_options)
           url = url_for(options)
 
-          if html_options
-            html_options = html_options.stringify_keys
-            href = html_options['href']
-            tag_options = tag_options(html_options)
-          else
-            tag_options = nil
-          end
+          href = html_options['href']
+          tag_options = tag_options(html_options)
 
           href_attr = "href=\"#{html_escape(url)}\"" unless href
           "<a #{href_attr}#{tag_options}>#{html_escape(name || url)}</a>".html_safe
-- 
1.7.1

